File: frame/src/main/java/com/dreampany/frame/data/source/repository/StateRepository.java
package com.dreampany.frame.data.source.repository;

import com.dreampany.frame.data.model.State;
import com.dreampany.frame.data.source.StateDataSource;
import com.dreampany.frame.misc.Room;
import com.dreampany.frame.misc.ResponseMapper;
import com.dreampany.frame.misc.RxMapper;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import hugo.weaving.DebugLog;
import io.reactivex.Maybe;

/**
 * Created by Hawladar Roman on 7/19/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
@Singleton
public class StateRepository extends Repository<Long, State> implements StateDataSource {

    private final StateDataSource room;

    @DebugLog
    @Inject
    StateRepository(RxMapper rx,
                    ResponseMapper rm,
                    @Room StateDataSource room) {
        super(rx, rm);
        this.room = room;
    }

    @Override
    public int getCount(long id, String type, String subtype) {
        return room.getCount(id, type, subtype);
    }

    @Override
    public boolean isEmpty() {
        return room.isEmpty();
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return Maybe.fromCallable(this::isEmpty);
    }

    @Override
    public int getCount(String type, String subtype, String state) {
        return room.getCount(type, subtype, state);
    }

    @Override
    public int getCount(String type, String subtype, String state, String substate) {
        return room.getCount(type, subtype, state, substate);
    }

    @Override
    public int getCount(long id, String type, String subtype, String state) {
        return room.getCount(id, type, subtype, state);
    }

    @Override
    public int getCount(long id, String type, String subtype, String state, String substate) {
        return room.getCount(id, type, subtype, state, substate);
    }

    @Override
    public Maybe<Integer> getCountRx(String type, String subtype, String state) {
        return room.getCountRx(type, subtype, state);
    }

    @Override
    public Maybe<Integer> getCountRx(long id, String type, String subtype, String state) {
        return room.getCountRx(id, type, subtype, state);
    }

    @Override
    public State getItem(long id, String type, String subtype, String state) {
        return room.getItem(id, type, subtype, state);
    }

    @Override
    public Maybe<State> getItemRx(long id, String type, String subtype, String state) {
        return room.getItemRx(id, type, subtype, state);
    }

    @Override
    public Maybe<State> getItemOrderByRx(String type, String subtype, String state, long from, long to) {
        return room.getItemOrderByRx(type, subtype, state, from, to);
    }

    @Override
    public int getCount() {
        return room.getCount();
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return room.getCountRx();
    }

    @Override
    public List<State> getItems(String type, String subtype) {
        return room.getItems(type, subtype);
    }

    @Override
    public List<State> getItems(long id, String type, String subtype) {
        return room.getItems(id, type, subtype);
    }

    @Override
    public State getItemOrderBy(String type, String subtype) {
        return null;
    }

    @Override
    public Maybe<State> getItemNotStateOrderByRx(String type, String subtype, String state) {
        return room.getItemNotStateOrderByRx(type, subtype, state);
    }

    @Override
    public Maybe<State> getItemOrderByRx(String type, String subtype) {
        return room.getItemOrderByRx(type, subtype);
    }

    @Override
    public Maybe<List<State>> getItemsRx(String type, String subtype) {
        return room.getItemsRx(type, subtype);
    }

    @Override
    public List<State> getItemsOrderBy(String type, String subtype, long from, long to) {
        return room.getItemsOrderBy(type, subtype, from, to);
    }

    @Override
    public Maybe<List<State>> getItemsOrderByRx(String type, String subtype, long from, long to) {
        return room.getItemsOrderByRx(type, subtype, from, to);
    }

    @Override
    public List<State> getItems(String type, String subtype, String state) {
        return null;
    }

    @Override
    public Maybe<List<State>> getItemsRx(String type, String subtype, String state) {
        return room.getItemsRx(type, subtype, state);
    }

    @Override
    public List<State> getItemsOrderBy(String type, String subtype) {
        return room.getItemsOrderBy(type, subtype);
    }

    @Override
    public Maybe<List<State>> getItemsOrderByRx(String type, String subtype) {
        return room.getItemsOrderByRx(type, subtype);
    }

    @Override
    public List<State> getItemsOrderBy(String type, String subtype, String state) {
        return room.getItemsOrderBy(type, subtype, state);
    }

    @Override
    public Maybe<List<State>> getItemsOrderByRx(String type, String subtype, String state) {
        return room.getItemsOrderByRx(type, subtype, state);
    }

    @Override
    public List<State> getItemsOrderBy(String type, String subtype, int limit) {
        return null;
    }

    @Override
    public Maybe<List<State>> getItemsOrderByRx(String type, String subtype, int limit) {
        return room.getItemsOrderByRx(type, subtype, limit);
    }

    @Override
    public List<State> getItemsOrderBy(String type, String subtype, String state, int limit) {
        return room.getItemsOrderBy(type, subtype, state, limit);
    }

    @Override
    public Maybe<List<State>> getItemsOrderByRx(String type, String subtype, String state, int limit) {
        return room.getItemsOrderByRx(type, subtype, state, limit);
    }

    @Override
    public List<State> getItems(String type, String subtype, int limit) {
        return room.getItems(type, subtype, limit);
    }

    @Override
    public Maybe<List<State>> getItemsRx(String type, String subtype, int limit) {
        return room.getItemsRx(type, subtype, limit);
    }

    @Override
    public boolean isExists(State state) {
        return room.isExists(state);
    }

    @Override
    public Maybe<Boolean> isExistsRx(State state) {
        return room.isExistsRx(state);
    }

    @Override
    public long putItem(State state) {
        return room.putItem(state);
    }

    @Override
    public Maybe<Long> putItemRx(State state) {
        return room.putItemRx(state);
    }

    @Override
    public List<Long> putItems(List<State> states) {
        return room.putItems(states);
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<State> states) {
        return room.putItemsRx(states);
    }

    @Override
    public State getItem(long id) {
        return room.getItem(id);
    }

    @Override
    public Maybe<State> getItemRx(long id) {
        return room.getItemRx(id);
    }

    @Override
    public List<State> getItems() {
        return room.getItems();
    }

    @Override
    public Maybe<List<State>> getItemsRx() {
        return room.getItemsRx();
    }

    @Override
    public List<State> getItems(int limit) {
        return room.getItems(limit);
    }

    @Override
    public Maybe<List<State>> getItemsRx(int limit) {
        return room.getItemsRx(limit);
    }
}

File: frame/src/main/kotlin/com/dreampany/framework/data/source/repository/StateRepository.kt
/*
package com.dreampany.frame.extra.source.repository

import com.dreampany.frame.extra.model.State
import com.dreampany.frame.extra.source.api.StateDataSource
import com.dreampany.frame.misc.ResponseMapper
import com.dreampany.frame.misc.Room
import com.dreampany.frame.misc.RxMapper
import io.reactivex.Maybe
import javax.inject.Inject
import javax.inject.Singleton

*/
/**
 * Created by Roman-372 on 7/19/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 *//*

@Singleton
class StateRepository @Inject constructor(
    rx: RxMapper,
    rm: ResponseMapper,
    @Room private val room: StateDataSource
): Repository<String, State>(rx, rm), StateDataSource {

    override fun getCountById(id: String, type: String, subtype: String): Int {
        return room.getCountById(id, type, subtype)
    }

    override fun getCountByIdRx(id: String, type: String, subtype: String): Maybe<Int> {
        return Maybe.empty()
    }

    override fun isEmpty(): Boolean {
        return room.isEmpty()
    }

    override fun isEmptyRx(): Maybe<Boolean> {
        return Maybe.fromCallable({ this.isEmpty() })
    }

    override fun getCount(type: String, subtype: String, state: String): Int {
        return room.getCount(type, subtype, state)
    }

    override fun getCountById(id: String, type: String, subtype: String, state: String): Int {
        return room.getCountById(id, type, subtype, state)
    }

    override fun getCountRx(type: String, subtype: String, state: String): Maybe<Int> {
        return room.getCountRx(type, subtype, state)
    }

    override fun getCountRx(id: String, type: String, subtype: String, state: String): Maybe<Int> {
        return room.getCountRx(id, type, subtype, state)
    }

    override fun getItem(id: String, type: String, subtype: String, state: String): State? {
        return room.getItem(id, type, subtype, state)
    }

    override fun getItemRx(id: String, type: String, subtype: String, state: String): Maybe<State> {
        return room.getItemRx(id, type, subtype, state)
    }

    override fun getItemOrderByRx(
        type: String,
        subtype: String,
        state: String,
        from: Long,
        to: Long
    ): Maybe<State> {
        return room.getItemOrderByRx(type, subtype, state, from, to)
    }

    override fun getCount(): Int {
        return room.getCount()
    }

    override fun getCountRx(): Maybe<Int> {
        return room.getCountRx()
    }

    override fun getItems(type: String, subtype: String): List<State>? {
        return room.getItems(type, subtype)
    }

    override fun getItemOrderBy(type: String, subtype: String): State? {
        return null
    }

    override fun getItemNotStateOrderByRx(
        type: String,
        subtype: String,
        state: String
    ): Maybe<State> {
        return room.getItemNotStateOrderByRx(type, subtype, state)
    }

    override fun getItemOrderByRx(type: String, subtype: String): Maybe<State> {
        return room.getItemOrderByRx(type, subtype)
    }

    override fun getItemsRx(type: String, subtype: String): Maybe<List<State>> {
        return room.getItemsRx(type, subtype)
    }

    override fun getItemsOrderBy(
        type: String,
        subtype: String,
        from: Long,
        to: Long
    ): List<State>? {
        return room.getItemsOrderBy(type, subtype, from, to)
    }

    override fun getItemsOrderByRx(
        type: String,
        subtype: String,
        from: Long,
        to: Long
    ): Maybe<List<State>> {
        return room.getItemsOrderByRx(type, subtype, from, to)
    }

    override fun getItem(type: String, subtype: String, state: String): State? {
        return room.getItem(type, subtype, state)
    }

    override fun getItemRx(type: String, subtype: String, state: String): Maybe<State> {
        return room.getItemRx(type, subtype, state)
    }

    override fun getItems(type: String, subtype: String, state: String): List<State>? {
        return room.getItems(type, subtype, state)
    }

    override fun getItemsRx(type: String, subtype: String, state: String): Maybe<List<State>> {
        return room.getItemsRx(type, subtype, state)
    }

    override fun getItemsOrderBy(type: String, subtype: String): List<State>? {
        return room.getItemsOrderBy(type, subtype)
    }

    override fun getItemsOrderByRx(type: String, subtype: String): Maybe<List<State>> {
        return room.getItemsOrderByRx(type, subtype)
    }

    override fun getItemsOrderBy(type: String, subtype: String, state: String): List<State>? {
        return room.getItemsOrderBy(type, subtype, state)
    }

    override fun getItemsOrderByRx(
        type: String,
        subtype: String,
        state: String
    ): Maybe<List<State>> {
        return room.getItemsOrderByRx(type, subtype, state)
    }

    override fun getItemsOrderBy(type: String, subtype: String, limit: Int): List<State>? {
        return null
    }

    override fun getItemsOrderByRx(type: String, subtype: String, limit: Int): Maybe<List<State>> {
        return room.getItemsOrderByRx(type, subtype, limit)
    }

    override fun getItemsOrderBy(
        type: String,
        subtype: String,
        state: String,
        limit: Int
    ): List<State>? {
        return room.getItemsOrderBy(type, subtype, state, limit)
    }

    override fun getItemsOrderByRx(
        type: String,
        subtype: String,
        state: String,
        limit: Int
    ): Maybe<List<State>> {
        return room.getItemsOrderByRx(type, subtype, state, limit)
    }

    override fun getItems(type: String, subtype: String, limit: Int): List<State>? {
        return room.getItems(type, subtype, limit)
    }

    override fun getItemsRx(type: String, subtype: String, limit: Int): Maybe<List<State>> {
        return room.getItemsRx(type, subtype, limit)
    }

    override fun isExists(state: State): Boolean {
        return room.isExists(state)
    }

    override fun isExistsRx(state: State): Maybe<Boolean> {
        return room.isExistsRx(state)
    }

    override fun putItem(state: State): Long {
        return room.putItem(state)
    }

    override fun putItemRx(state: State): Maybe<Long> {
        return room.putItemRx(state)
    }

    override fun putItems(states: List<State>): List<Long>? {
        return room.putItems(states)
    }

    override fun putItemsRx(states: List<State>): Maybe<List<Long>> {
        return room.putItemsRx(states)
    }

    override fun delete(state: State): Int {
        return room.delete(state)
    }

    override fun deleteRx(state: State): Maybe<Int> {
        return Maybe.empty()
    }

    override fun delete(states: List<State>): List<Long>? {
        return null
    }

    override fun deleteRx(states: List<State>): Maybe<List<Long>> {
        return Maybe.empty()
    }

    override fun getItem(id: String): State? {
        return null
    }

    override fun getItemRx(id: String): Maybe<State> {
        return Maybe.empty()
    }

    override fun getItems(): List<State>? {
        return room.getItems()
    }

    override fun getItemsRx(): Maybe<List<State>> {
        return room.getItemsRx()
    }

    override fun getItems(limit: Int): List<State>? {
        return room.getItems(limit)
    }

    override fun getItemsRx(limit: Int): Maybe<List<State>> {
        return room.getItemsRx(limit)
    }
}*/
